### PR TITLE
feat(testing): update cypress to 9.x

### LINKED ISSUE #7954: Support Cypress 9
## Description

The next major version of Cypress, Cypress 9, has been released. Nx should officially support Cypress 9 and make any necessary updates.

## Motivation

Nx should support the latest Cypress version, as it includes
- security fixes
- bug fixes
- new features

## Suggested Implementation

See https://github.com/nrwl/nx/issues/6430 for the work supporting the last Cypress major version (Cypress 8), substituting for `9.1.0` and `9`.

> - Follow #5624 as an example.
>     - Update current Cypress version in this repo to `8.0.0` or newer (if newer version is released).
>     - Update current supported Cypress versions to include all `8` versions (`< 9` in `peerDependencies`)
> - Make sure to check the changes to Cypress CLI config types and ensure the Nx Cypress CLI is compatible or makes any necessary changes.

## Alternate Implementations

N/A

### PR DESCRIPTION
**Pull request by puku0x**
## Current Behavior

Cypress v8.x is supported.

## Expected Behavior

Cypress v9.x is supported.

## Related Issue(s)

Closes #7954
From v7 to v8, see https://github.com/nrwl/nx/pull/6808

### CODE REVIEW

**Comment by puku0x** [comment]
There are several breaking changes in v9, for example, Win32 is no more supported.
https://docs.cypress.io/guides/references/changelog#9-0-0

🤔Should upgrading Cypress be treated as a breaking change of Nx? (If so, I will close this PR)

File: package.json [hunk 1]
```diff
@@ -130,7 +130,7 @@
     "copy-webpack-plugin": "^9.0.1",
     "core-js": "^3.6.5",
     "css-minimizer-webpack-plugin": "^3.1.1",
-    "cypress": "^8.3.0",
+    "cypress": "^9.1.0",
```
Comment (puku0x) [inline package.json:133]:
```
yarn upgrade cypress@^9.1.0
```

**Review by barbados-clemens — APPROVED** [review]
Everything looks good 👍
locally tests and verified changes

Thanks for the contribution 😄 

Note: You'll probably want to rebase with master and push to fix the failing build.

**Comment by puku0x** [comment]
Locally tested and passed. https://nx.app/runs/EIoLtT1WVbO

**Comment by puku0x** [comment]
Finally, CI passed😊

**Pull request merged**